### PR TITLE
Transaction support

### PR DESCRIPTION
**Pull request by negator**
Adds transaction support via two new functions added to the go-sdk:
https://docs.aws.amazon.com/sdk-for-go/api/service/dynamodb/#DynamoDB.TransactGetItemsWithContext

https://docs.aws.amazon.com/sdk-for-go/api/service/dynamodb/#DynamoDB.TransactWriteItemsWithContext

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -36,9 +36,9 @@ go_test(
 #### Atlassian / Localstack ####
 genrule(
     name = "dynamodb",
-    outs = ["localstack-run-id"],
-    cmd = "docker stop localstack || true; docker rm -f localstack || true; docker run -d -p 4567-4576:4567-4576 --name localstack atlassianlabs/localstack:0.4.1 > $@",
+    outs = ["dynamodb-local-run-id"],
+    cmd = "docker stop dynamodb-local || true; docker rm -f dynamodb-local || true; docker run -d -p 4569:8000 --name dynamodb-local amazon/dynamodb-local:latest > $@",
```
Comment (brendanjryan) [inline BUILD]: 👍 I trust that this image is better than the localstack (dynalite) one?

File: domino.go [hunk 2]
```diff
@@ -668,6 +772,211 @@ func (o *putOutput) Result(item interface{}) (err error) {
...
+
+	delayed := func() error {
+		if d.batches == nil {
+			d.batches = []*dynamodb.TransactWriteItemsInput{}
+		}
+		batchLen := len(d.batches)
+
+		var batch *dynamodb.TransactWriteItemsInput
+		// Create a new batch if there are none, or the last batch >= 10
+		if batchLen == 0 || len(d.batches[batchLen-1].TransactItems) >= 10 {
```
Comment (brendanjryan) [inline domino.go]: it seems like we use this `10` number in a lot of places - could we possibly make this into a packge-level constant

File: domino_test.go [hunk 3]
```diff
@@ -455,7 +543,7 @@ func TestExpressions(t *testing.T) {
 	expectedFilter := "registrationDate = :filter_1 OR contains(lastName,:filter_2) OR (NOT registrationDate = :filter_3) OR (size(visits) <=:filter_4 AND size(firstName) >=:filter_5) OR registrationDate = :filter_6 OR registrationDate <= :filter_7 OR (registrationDate between :filter_8 and :filter_9) OR (registrationDate in (:filter_10,:filter_11))"
 	assert.Equal(t, expectedFilter, *q.Build().FilterExpression)
 
-	fmt.Println(*q.Build())
+	// fmt.Println(*q.Build())
```
Comment (brendanjryan) [inline domino_test.go]: can we remove this?

**Review by brendanjryan — APPROVED** [review]
LGTM - sorry for the delay on this!

File: BUILD (hunk 1 above)
Comment (promiseofcake) [reply BUILD]: I think this is "officially" provided by Amazon so I would hope so.

**Comment by negator** [comment]
DynamoDB local does not have support for transactions yet, which makes this difficult to unit test. However it seems it will be added soon: https://forums.aws.amazon.com/thread.jspa?threadID=294782

Going to delay merging this until the officially supported version of DynamoDBLocal supports them

**Comment by negator** [comment]
bout time https://aws.amazon.com/about-aws/whats-new/2019/02/amazon-dynamodb-local-adds-support-for-transactional-apis-on-demand-capacity-mode-and-20-gsis/

**Pull request merged**